lib/views/pay/pay_view.dart

```diff
@@ -1,114 +1,255 @@
+                                      side: const BorderSide(
+                                        color: Colors.white,
+                                        width: 2.0,
+                                      ),
+                                    ),
+                                    margin: const EdgeInsets.only(top: 15),
+                                    child: Padding(
+                                      padding: const EdgeInsets.all(15),
+                                      child: Center(
+                                        child: Text(
+                                          invoice.invoice.description,
+                                          style: const TextStyle(fontSize: 18),
+                                        ),
+                                      ),
+                                    ),
+                                  ),
+                                ),
+                                const SizedBox(
+                                  height: 10,
+                                ),
+                                Expanded(
+                                  flex: 1,
+                                  child: Row(
+                                    mainAxisAlignment:
+                                        MainAxisAlignment.spaceBetween,
+                                    children: [
+                                      Text(
+                                        'Created time \n${invoice.invoice.createdTime}',
+                                        style: const TextStyle(fontSize: 18),
+                                      ),
+                                      Text(
+                                        'Expiration time \n${invoice.invoice.expirationTime}',
+                                        style: const TextStyle(fontSize: 18),
+                                      ),
+                                    ],
+                                  ),
+                                )
+                              ],
+                            ),
+                          ),
+                        ),
+                      ),
+                    ),
+                  ),
+                  invoice.invoice.amount != 0
+                      ? Expanded(
+                          flex: 1,
+                          child: MainCircleButton(
+                              width: 200,
+                              icon: Icons.paypal,
+                              label: 'Pay invoice',
+                              onPress: () async {
+                                await payInvoice(boltString!);
+                              }),
+                        )
+                      : const Expanded(
+                          flex: 1,
+                          child: Text(
+                              "You need to specify the amount of the given invoice"),
+                        ),
+                ],
+              ));
+        });
+  }
```

Comment:
It is possible to put the Widged inside the bottom dialog in a separate file? in this way we can reuse it somewhere else